spec.html

```diff
@@ -6348,52 +6348,58 @@ <h1>
         1. Return ~unused~.
       </emu-alg>
       <emu-note>
-        <p>This abstract operation creates a property whose attributes are set to the same defaults used for built-in methods and methods defined using class declaration syntax. Normally, the property will not already exist. If it does exist and is not configurable or if _O_ is not extensible, [[DefineOwnProperty]] will return *false*.</p>
+        <p>This abstract operation creates a property whose attributes are set to the same defaults used for built-in methods and methods defined using class declaration syntax. Normally, the property will not already exist. If it does exist, [[DefineOwnProperty]] is guaranteed to complete normally and return *true* within this specification.</p>
+      </emu-note>
+      <emu-note>
+        <p>This abstract operation is functionally equivalent to CreateNonEnumerableDataProperty, they only differ semantically.</p>
       </emu-note>
     </emu-clause>
 
-    <emu-clause id="sec-createdatapropertyorthrow" type="abstract operation">
+    <emu-clause id="sec-createnonenumerabledataproperty" oldids="sec-createnonenumerabledatapropertyorthrow" type="abstract operation">
       <h1>
-        CreateDataPropertyOrThrow (
+        CreateNonEnumerableDataProperty (
           _O_: an Object,
           _P_: a property key,
           _V_: an ECMAScript language value,
-        ): either a normal completion containing a Boolean or a throw completion
+        ): ~unused~
       </h1>
       <dl class="header">
         <dt>description</dt>
-        <dd>It is used to create a new own property of an object. It throws a *TypeError* exception if the requested property update cannot be performed.</dd>
+        <dd>It is used to create a new non-enumerable own property of an ordinary object.</dd>
       </dl>
       <emu-alg>
-        1. Let _success_ be ? CreateDataProperty(_O_, _P_, _V_).
-        1. If _success_ is *false*, throw a *TypeError* exception.
-        1. Return _success_.
+        1. Assert: _O_ is an ordinary, extensible object with no non-configurable properties.
```

Comment:
How can this assertion be made? This spec isn’t the only thing that calls these AOs.

Reply:
I didn't make that change, it only shows as new from the re-ordering. This is already in the current spec.